### PR TITLE
Feature, FileSelectorView

### PR DESCRIPTION
**Pull request by EAlexisDuque**
New CustomComponent `FileSelectorView`

### CODE REVIEW

File: app/src/main/java/co/condorlabs/customcomponents/fileselector/FileSelectorView.kt [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                        result.resume(which)
+                    }
+                    .create()
+                    .show()
+        }
+
+        return suspendCoroutine { continuation -> result = continuation }
+    }
+
+    fun setOnClickListener(l: OnClickListener) {
```
Comment (leidyzulu) [inline app/src/main/java/co/condorlabs/customcomponents/fileselector/FileSelectorView.kt]: rename "l"

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/FileSelectorViewTest.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        //Given
+        val view = onView(withId(R.id.myCustomLayoutSelector))
+
+        //When
+        view.perform(click())
+
+        //Then
+        //withHint("Phone").matches(view)
+    }
+}
```
Comment (leidyzulu) [inline test/src/androidTest/java/co/condorlabs/customcomponents/test/FileSelectorViewTest.kt]: add blank line

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/FileSelectorViewTest.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        restartActivity()
+
+        //Given
+        val view = onView(withId(R.id.myCustomLayoutSelector))
+
+        //When
+        view.perform(click())
+
+        //Then
+        //withHint("Phone").matches(view)
```
Comment (leidyzulu) [inline test/src/androidTest/java/co/condorlabs/customcomponents/test/FileSelectorViewTest.kt]: ?

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/FileSelectorViewTest.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+@RunWith(AndroidJUnit4::class)
+class FileSelectorViewTest : MockActivityTest() {
+
+    @Before
+    fun setup() {
+        MockActivity.layout = R.layout.activity_file_selector_view_test
+    }
+
+    @Test
```
Comment (leidyzulu) [inline test/src/androidTest/java/co/condorlabs/customcomponents/test/FileSelectorViewTest.kt]: add test to validate title, subtitle, and validate the dialog

File: app/src/main/java/co/condorlabs/customcomponents/fileselectorview/FileSelectorView.kt [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+    override fun onAttachedToWindow() {
+        super.onAttachedToWindow()
+        setup()
+    }
+
+    override fun isValid(): ValidationResult {
+        if (mIsRequired && mFileSelectorValue == null) {
+            return ValidationResult(false, context.getString(R.string.file_selector_default_error))
+
```
Comment (leidyzulu) [inline app/src/main/java/co/condorlabs/customcomponents/fileselectorview/FileSelectorView.kt]: delete blank line

File: app/src/main/java/co/condorlabs/customcomponents/fileselectorview/FileSelectorView.kt [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+    override fun onClick(v: View?) {
+        showFileSelectorDialog()
+    }
+
+    override fun getValue(): FileSelectorValue? {
+        return mFileSelectorValue
+    }
+
+    override fun getErrorValidateResult(): ValidationResult {
+        return ValidationResult(false, "You should attach a file")
```
Comment (leidyzulu) [inline app/src/main/java/co/condorlabs/customcomponents/fileselectorview/FileSelectorView.kt]: create stringresource

File: app/src/main/java/co/condorlabs/customcomponents/fileselectorview/FileSelectorView.kt [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+                    context.getString(R.string.photo_string)
+                )
+            ) { dialog, which ->
+                dialog.dismiss()
+                result.resume(which)
+            }
+            .create()
+            .show()
+
+
```
Comment (leidyzulu) [inline app/src/main/java/co/condorlabs/customcomponents/fileselectorview/FileSelectorView.kt]: delete blank line

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/FileSelectorViewTest.kt [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+    fun shouldChangeTitle() {
+        MockActivity.layout = R.layout.activity_file_selector_title
+        restartActivity()
+
+        //Given
+        val view = Espresso.onView(withText("This is the Title"))
+
+
```
Comment (leidyzulu) [inline test/src/androidTest/java/co/condorlabs/customcomponents/test/FileSelectorViewTest.kt]: delete blank line

File: app/src/main/java/co/condorlabs/customcomponents/fileselectorview/FileSelectorView.kt [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+        val attrsArray = context.obtainStyledAttributes(attributes, R.styleable.FileSelectorView)
+
+        if (attrsArray.hasValue(R.styleable.FileSelectorView_dialog_title)) {
+            attrsArray.getString(R.styleable.FileSelectorView_dialog_title).let { dialogTitle ->
+                mDialogTitle = dialogTitle
+            }
+        }
+
+        if (attrsArray.hasValue(R.styleable.FileSelectorView_dialog_options)) {
+            attrsArray.getResourceId(R.styleable.FileSelectorView_dialog_options, -1).let { optionsArrayResourceId ->
```
[the remark this answers is not in the document]
Comment (ogallon) [reply app/src/main/java/co/condorlabs/customcomponents/fileselectorview/FileSelectorView.kt]: DONE

File: app/src/main/java/co/condorlabs/customcomponents/fileselectorview/FileSelectorView.kt (hunk 5 above)
Comment (ogallon) [reply app/src/main/java/co/condorlabs/customcomponents/fileselectorview/FileSelectorView.kt]: done

File: app/src/main/java/co/condorlabs/customcomponents/fileselectorview/FileSelectorView.kt (hunk 6 above)
Comment (ogallon) [reply app/src/main/java/co/condorlabs/customcomponents/fileselectorview/FileSelectorView.kt]: done

File: app/src/main/java/co/condorlabs/customcomponents/fileselectorview/FileSelectorView.kt (hunk 7 above)
Comment (ogallon) [reply app/src/main/java/co/condorlabs/customcomponents/fileselectorview/FileSelectorView.kt]: done

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/FileSelectorViewTest.kt (hunk 8 above)
Comment (ogallon) [reply test/src/androidTest/java/co/condorlabs/customcomponents/test/FileSelectorViewTest.kt]: done

**Pull request merged**